pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -302,20 +307,35 @@ func (c *Controller) syncService(key string) error {
 	return nil
 }
 
-// obj could be a *v1.Service or a DeletionalFinalStateUnknown marker item
+// obj could be a *v1.Service.
 func (c *Controller) enqueueService(obj interface{}) {
+	if key, err := enqueueKey(obj, c.queue); err == nil {
+		_ = c.serviceSelectorCache.Update(key, obj.(*v1.Service))
+	}
+}
+
+// obj could be a DeletionalFinalStateUnknown marker item.
+func (c *Controller) dequeueService(obj interface{}) {
```

Comment:
Maybe this should be `onServiceDelete`? The corresponding comment could be `onServiceDelete removes the Service Selector from the cache and queues the Service for processing.`